### DISCUSSION TITLE
Comparison of cooperativity and competition across communities

### DISCUSSION THREAD

**Discussion by cfrioux** [micom-dev/micom#126] [Q&A] [answered]
Hello! I am interested in comparing potentials for cooperativity and competition in several communities using micom. 
I think `knockout_taxa` is a key to answer this question, as the [documentation](https://micom-dev.github.io/micom/taxa_knockouts.html) says that

> If the knowckout of one bacteria increases the growth of another one this indicates competitivity whereas a dimished growth rate indicted cooperativity. 

In large communities, this analysis provides a large number of values, and comparing two or more communities that have distinct composition gets tricky. I was therefore wondering whether we could find out a way to get a unique score by community, similar to what [SMETANA](https://www.pnas.org/doi/10.1073/pnas.1421834112) could do for instance. 

Do you have any clue in this direction?  The more negative values we obtain in the output table, the more cooperativity could occur among species, would it make sense to sum the values?
Thanks.

**Comment by cdiener** [comment, marked as answer]
Hi, that is a great question and something where MICOM is definitely lacking a bit. What you are proposing makes sense to me. It would be pretty different from SMETANA since it does assume some kind of optimality and only uses the "best" solution to derive the effect. It should work in general though and a strong positive mean would indicate competition whereas as strong negative mean would indicate mutualism or exploitation (using the mean here to account for different number of taxa in communities). I would suspect that overall the scores are shifted towards positive because currently the knockout method does not readjust the relative abundances, meaning that all resources taken up by the removed taxon are available for other taxa. It might also be somewhat slow because it needs to run $n_{taxa}$ optimizations for each sample.

That said, I definitely think that all the SMETANA scores should be added to MICOM. Some might only be applicable to smaller models due to the complexity in calculating them (like the full SCS-MUS-MPS score) but they need to be added for sure. I already have some new features slated for inclusion to quantify metabolic interactions between taxa and will work on adding those too. Might take me a while, so, as always, I would :heart: PRs in case you or somebody else would like to work on that.

Congrats on the paper btw :tada:

**Comment by cfrioux** [comment]
Thanks Christian for your insights. It would indeed be different from SMETANA as scores would depend on the habitat in MICOM. I'll give it a try with the sum and the mean!

Nice preview article :)

**Reply by cdiener** [reply]
Doesn't SMETANA also require you to specify a medium? Or was that only Daniel's implementation?

**Reply by cfrioux** [reply]
It is an option. If I understand correctly : if provided, the medium will be used to design a minimal medium. Otherwise, the minimal medium will be calculated from the set of exchange reactions. Minimal medium in an interacting and non-interacting community are an input for MIP and MRO calculation. So the medium is taken into account but what matters is the minimal medium derived out of it. 

What is indicated in the original paper (PNAS 2015): 

> MRO is an intrinsic property of any community, independent of the habitat and robust in view of multiple minimal nutritional requirements for any (or all) member species, and providing an upper bound on the overall degree of resource competition. 

And in Daniel's [recent paper](https://www.nature.com/articles/s41559-020-01353-4) 

> MRO and MIP estimate the respective interaction metrics at their theoretical limit and therefore do not require information on the resources actually available in the habitat. The operating degree of competition and cooperation in a given community will therefore be habitat dependent. The context-independent nature of MRO and MIP makes them suitable for application to co-occurring communities spanning multiple and diverse habitats.
